test/inductor/test_torchinductor.py

```diff
@@ -4588,6 +4609,14 @@ def fn(x):
                 re.search(pattern, code), msg="Found bad index_expr in code:\n" + code
             )
 
+    def test_float_index_expression_type_promotion(self):
+        # Test that float indexing expressions participate in type promotion
+        def fn(x):
+            return x + 1.0 / x.size(0)
```

Comment:
If `x` is not defined inside `fn`, `x` will not be seen as an indexing expression

Reply:
`x.size(0)` is the indexing expression here, `x` is just a tensor.